src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -301,32 +283,24 @@ private boolean validateTargetPort(String targetPort) {
      * screen.
      */
     public FormValidation doCheckTargetPort(@QueryParameter("targetPort") final String targetPort) {
-        if (!validateTargetPort(targetPort)) {
+        if (!validatePort(targetPort)) {
             return FormValidation.error("Invalid Port");
         }
 
         return FormValidation.ok("Valid Port");
     }
 
-    private boolean validateTargetLogCollectionPort(String targetLogCollectionPort) {
-        if(!DatadogClient.ClientType.DSD.name().equals(reportWith) || !collectBuildLogs) {
-            return true;
-        }
-
-        return StringUtils.isNotBlank(targetLogCollectionPort) && StringUtils.isNumeric(targetLogCollectionPort);
-    }
-
     /**
      * @param targetLogCollectionPort - The Log Collection Port which the plugin will report to.
      * @return a FormValidation object used to display a message to the user on the configuration
      * screen.
      */
     public FormValidation doCheckTargetLogCollectionPort(@QueryParameter("targetLogCollectionPort") final String targetLogCollectionPort) {
-        if (!validateTargetLogCollectionPort(targetLogCollectionPort)) {
-            return FormValidation.error("Invalid Port");
+        if (!validatePort(targetLogCollectionPort) && collectBuildLogs) {
```

Comment:
Did you fix a bug here?

Reply:
No- validateTargetLogCollectionPort used to check for type of Client, valid port, and collectBuildLogs. I moved the port checking to validatePort so that it could be used in more cases, but had to include the collectBuildLogs check in this case.